tracer/src/Datadog.Trace.Coverage.collector/AssemblyProcessor.cs

```diff
@@ -163,11 +196,30 @@ public void Process()
                     if (moduleTypeMethods.Count > 0)
                     {
                         moduleCoverageMetadataImplCtor.Body.Instructions.Add(Instruction.Create(OpCodes.Ldc_I4, moduleTypeMethods.Count));
-                        moduleCoverageMetadataImplCtor.Body.Instructions.Add(Instruction.Create(OpCodes.Newarr, module.ImportReference(typeof(int))));
+                        moduleCoverageMetadataImplCtor.Body.Instructions.Add(Instruction.Create(OpCodes.Newarr,  module.TypeSystem.Int32));
                     }
                     else
                     {
-                        var arrayEmptyOfIntMethodReference = module.ImportReference(ArrayEmptyOfIntMethod);
+                        // Emit Array.Empty<int>() call
+                        if (arrayEmptyOfIntMethodReference is null)
+                        {
+                            var assemblyReferences = module.AssemblyReferences.ToArray();
+                            arrayEmptyOfIntMethodReference = (GenericInstanceMethod)module.ImportReference(ArrayEmptyOfIntMethod);
+                            arrayEmptyOfIntMethodReference.DeclaringType.Scope = module.TypeSystem.CoreLibrary;
+                            arrayEmptyOfIntMethodReference.GenericArguments[0] = module.TypeSystem.Int32;
+
+                            // If the `ImportReference` sentence add a new assembly reference (cross runtime versions)
+                            // we revert the reference list at the previous state.
+                            if (assemblyReferences.Length != module.AssemblyReferences.Count)
+                            {
+                                module.AssemblyReferences.Clear();
+                                foreach (var assemblyReference in assemblyReferences)
+                                {
+                                    module.AssemblyReferences.Add(assemblyReference);
+                                }
+                            }
```

Comment:
I'll take your word for it that this is safe 😅

Reply:
It's a simple AssemblyRef list but the index on the list is important, that's why I do this just after the `ImportReference` statement.